src/Sylius/Behat/Context/Api/Shop/ProductContext.php

```diff
@@ -600,16 +600,50 @@ public function iShouldNotBeAbleToSelectTheVariant(string $productVariantName):
 
     /**
      * @Then /^I should(?:| also) see the product association "([^"]+)" with (products "[^"]+" and "[^"]+")$/
+     * @Then /^I should(?:| also) see the product association "([^"]+)" with (product "[^"]+")$/
      */
-    public function iShouldSeeTheProductAssociationWithProductsAnd(string $productAssociationName, array $products): void
+    public function iShouldSeeTheProductAssociationWithProductsAnd(string $productAssociationName, array|ProductInterface $products): void
+    {
+        if (!is_array($products)) {
+            $products = [$products];
+        }
+
+        Assert::true($this->isProductAssociationWithProductsAvailable($productAssociationName, $products));
+    }
+
+    /**
+     * @Then /^I should(?:| also) not see the product association "([^"]+)" with (product "[^"]+")$/
+     */
+    public function iShouldNotSeeTheProductAssociationWithProduct(string $productAssociationName, ProductInterface $product): void
+    {
+        Assert::false($this->isProductAssociationWithProductsAvailable($productAssociationName, [$product]));
+    }
+
+    /**
+     * @Then /^I should not see the product association "([^"]+)"$/
+     */
+    public function iShouldNotSeeTheProductAssociationWithProducts(string $productAssociationName): void
     {
         /** @var ProductInterface $product */
         $product = $this->sharedStorage->get('product');
 
         $response = $this->client->show(Resources::PRODUCTS, $product->getCode());
         $associations = $this->responseChecker->getValue($response, 'associations');
 
-        Assert::true($this->hasAssociationsWithProducts($associations, $productAssociationName, $products));
+        $associationNames = [];
+
+        foreach ($associations as $association) {
+            $associationResponse = $this->client->showByIri($association);
+
+            if ($associationResponse->getStatusCode() === 200) {
+                $associationTypeIri = $this->responseChecker->getValue($associationResponse, 'type');
+                $associationTypeResponse = $this->client->showByIri($associationTypeIri);
+
+                $associationNames[] = $this->responseChecker->getValue($associationTypeResponse, 'name');
+            }
```

Comment:
I wonder if this "if" is needed
```suggestion
            $associationTypeIri = $this->responseChecker->getValue($associationResponse, 'type');
            $associationTypeResponse = $this->client->showByIri($associationTypeIri);

            $associationNames[] = $this->responseChecker->getValue($associationTypeResponse, 'name');
```

Reply:
There is a problem when association is not found, the `Expected 'type' not found.` occurs becouse of `$associationTypeIri = $this->responseChecker->getValue($associationResponse, 'type');` line here